zeebe/gateway-protocol/src/main/proto/rest-api.yaml

```diff
@@ -4862,7 +4831,14 @@ components:
           type: string
           description: The ID of the document.
         metadata:
-          $ref: "#/components/schemas/DocumentMetadata"
+          description: sjh
+          # sjhsjhsjh -- this is different from the source!
+          #   it needs to use allOf to consume DocumentMetadata.
+          #   I'm not sure why, because the oas spec seems to say applying the $ref without allOf is fine...
+          #   but we did allOf in every other schema for some reason, so this is the only one that failed.
+          type: object
+          allOf:
+            - $ref: "#/components/schemas/DocumentMetadata"
```

Comment:
oh boy, this was an adventure. 

* I was getting weird crashes from vacuum with my new rule added, on these properties that pointed directly at a `#ref`. 
   * The crash message was `panic: interface conversion: *base.ObjectReference is not base.Foundational: missing method AddEdge`
* To track it down I removed the entire spec and added it back in a little at a time until I identified the culprit...that's how I found the objects that weren't used (mentioned in a previous comment).
* I don't know _why_ my rule was crashing on these properties. The rule finds all property nodes, and then checks for the existence of a `description` property, none of which seems to relate to the `$ref`. 
* I don't think that the direct `$ref` (the left side of this diff) is a violation of the OpenAPI spec. 
* But weirdly, we do this `allOf` inheritance instead of direct `$ref` for most properties already anyway, and the `allOf` inheritance works fine with the rule....
* Since there are only a few cases where we are using direct `$ref` instead of `allOf` inheritance, it seems reasonable to just change those few cases.
* My big concern is that if someone attempts to add a direct `$ref`, the output from vacuum is not helpful at all.

Reply:
ok, I figured out why this was happening, or at least, how this example was different from others.

1. If a property inherits another schema via `allOf`, then a `description` is required on the property, or else the new rule will fail (clearly) for this property. 
2. If a property inherits another schema via a direct `$ref`, then no `description` is required on the _property_. A `description` _is_ required on the inherited schema, though, and if it doesn't exist, the vacuum rule will crash. 

I actually think that direct `$ref` inheritance is more accurate in describing the spec, so I've changed many instances of `allOf` inheritance to direct `$ref` inheritance. _However..._ this spreads a pattern in the spec that _could_ result in a vacuum crash, which I don't feel great about. I didn't consider this until after I'd already started converting to direct`$ref`s though, and I wanted to be consistent with the changes I was making, so that's where I ended up. We should discuss this tradeoff though, and choose a preference.

Reply:
So, this vacuum behavior makes a lot of sense in general. If you use a `$ref`, all sibling attributes are ignored because a ref is meant to be a drop-in that will be expanded in this place and "overrides" whatever was defined as siblings to it. Thus, the description must sit in the `$ref` as well.

If you go for an `allOf`, you must provide the description at the property itself. The type of the property described by an `allOf` is basically an anonymous, custom type that needs a description itself.

So, I would go about it this way:
* If there is a dedicated description at the property, referencing one `$ref` in an `allOf`, keep it as is because that property wants to define a dedicated description, overriding that of the `$ref`.
* If there is no description at an `allOf` property using one `$ref`, make it a direct `$ref` and use the description from that reference.

Let me know if that makes sense or if I overlooked anything.

Reply:
What you're saying makes sense, and I _think_ we're in alignment in terms of when to use which. 

I also _think_ this means I shouldn't revert any of my changes, because I don't see any where it seems more semantic to describe the schema property instead of the inherited component. I may have misinterpreted.

Reply:
Heads up that I found [an issue in how the docs generator handles these two different approaches](https://github.com/camunda/camunda-docs/issues/4610). I think I'm changing my opinion, to always use `allOf`, because it doesn't wreck the docs right now 😅. 

I'm going to come back and update in this PR.

Reply:
PR is updated to revert these changes!

Reply:
🚀